python/hidet/graph/ops/definitions/matmul/matmul_f32_x86.py

```diff
@@ -0,0 +1,411 @@
+                    aa = avx_f32x4_broadcast(~a[2, pp])
+                    c2 = avx_f32x4_fmadd(aa, bb, c2)
+                    aa = avx_f32x4_broadcast(~a[3, pp])
+                    c3 = avx_f32x4_fmadd(aa, bb, c3)
+                avx_f32x4_store(~c[0, 0], c0)
+                avx_f32x4_store(~c[1, 0], c1)
+                avx_f32x4_store(~c[2, 0], c2)
+                avx_f32x4_store(~c[3, 0], c3)
+
+            micro_kernel = micro_kernel_6x16
+            if tile_m == 8 and tile_n == 8:
+                micro_kernel = micro_kernel_8x8
+            elif tile_m == 4 and tile_n == 8:
+                micro_kernel = micro_kernel_4x8
+            elif tile_m == 4 and tile_n == 4:
+                micro_kernel = micro_kernel_4x4
+
+            @hidet.script
+            def macro_kernel(a: packed_a_type, b: packed_b_type, c_in_macro: c_type, ib: int32, jb: int32, pb: int32):
+                mpanels = (ib + tile_m - 1) // tile_m
+                npanels = (jb + tile_n - 1) // tile_n
+                _mr = ib % tile_m
+                _nr = jb % tile_n
+
+                # Loop 2
+                para = 'p' + str(nthreads)
+                for mpanel in grid(mpanels, attrs=para):
+                    mr = tile_m if mpanel != mpanels - 1 or _mr == 0 else _mr
+                    ii = mpanel * tile_m
+                    # Loop 1
+                    for npanel in range(npanels):
+                        nr = tile_n if npanel != npanels - 1 or _nr == 0 else _nr
+                        jj = npanel * tile_n
+                        # micro-kernel
+                        if mr == tile_m and nr == tile_n:
+                            micro_kernel(~a[ii, 0], ~b[0, jj], ~c_in_macro[ii, jj], pb, m_size, n_size)
+                        else:
+                            temp_c = tensor(
+                                scope=DeclareScope.Default, dtype='float32', layout=row_layout(tile_m, tile_n)
+                            )
+                            for tempi in range(tile_m):
+                                for tempj in range(tile_n):
+                                    temp_c[tempi, tempj] = 0.0
+                            micro_kernel(~a[ii, 0], ~b[0, jj], temp_c, pb, tile_m, tile_n)
+                            for remain_row, remain_col in grid(mr, nr):
+                                c_in_macro[ii + remain_row, jj + remain_col] += temp_c[remain_row, remain_col]
+
+            @hidet.script
+            def matmul_kernel_x86(a_ptr: ~float32, b_ptr: ~float32, c_ptr: ~float32):
+                a = as_tensor_pointer(a_ptr, dtype=float32, shape=[m_size, k_size])
+                b = as_tensor_pointer(b_ptr, dtype=float32, shape=[k_size, n_size])
+                c = as_tensor_pointer(c_ptr, dtype=float32, shape=[m_size, n_size])
+                mbs = (m_size + block_m - 1) // block_m
+                nbs = (n_size + block_n - 1) // block_n
+                kbs = (k_size + block_k - 1) // block_k
```

Comment:
If we declare kernel function in this way (use pointer as input instead of directly declare the input tensor), we will not support prologue/epilogue fusion. In this case, we should override the `allow_prologue(...)` and `allow_epilogue(...)` to return False. A better way is to disable prologue fusion and enable epilogue fusion, and decalre c as the tensor directly in the parameter list.